### PR TITLE
Fix date validator to work with new pattern widget output (new)

### PR DESCRIPTION
**Pull request by keul**
See https://github.com/smcmahon/Products.PloneFormGen/pull/165

### CODE REVIEW

**Comment by mauritsvanrees** [comment]
Thanks for working on this! It looks good on new forms.

But it fails when I try it on existing forms:
- With master, create a form with a date field. This looks ugly, so we indeed need a fix.
- Switch to the pull request branch.
- The date widget still looks ugly.
- When I quick edit the date widget and save it, I get 'there was an error submitting the form'.
- The javascript console says 'unable to load resource .../form/some-date/base-edit'. I can open that field edit page just fine though.
- Viewing the form fails:

```
http://localhost:8080/Plone5en/form/fg_base_view_p3
Traceback (innermost last):
  Module ZPublisher.Publish, line 138, in publish
  Module ZPublisher.mapply, line 77, in mapply
  Module ZPublisher.Publish, line 48, in call_object
  Module Products.CMFFormController.FSControllerPageTemplate, line 91, in __call__
  Module Products.CMFFormController.BaseControllerPageTemplate, line 32, in _call
  Module Shared.DC.Scripts.Bindings, line 322, in __call__
  Module Shared.DC.Scripts.Bindings, line 359, in _bindAndExec
  Module Products.CMFCore.FSPageTemplate, line 237, in _exec
  Module Products.CMFCore.FSPageTemplate, line 177, in pt_render
  Module Products.PageTemplates.PageTemplate, line 87, in pt_render
  Module zope.pagetemplate.pagetemplate, line 132, in pt_render
  Module five.pt.engine, line 98, in __call__
  Module z3c.pt.pagetemplate, line 163, in render
  Module chameleon.zpt.template, line 261, in render
  Module chameleon.template, line 191, in render
  Module chameleon.template, line 171, in render
  Module 0bbf6f6defcbf8c016f26de1d3bd541d.py, line 977, in render
  Module eae13f840a318c6383aa1ce8ed61ec23.py, line 1223, in render_master
  Module eae13f840a318c6383aa1ce8ed61ec23.py, line 420, in render_content
  Module 0bbf6f6defcbf8c016f26de1d3bd541d.py, line 965, in __fill_content_core
  Module 0bbf6f6defcbf8c016f26de1d3bd541d.py, line 332, in render_content_core
  Module e94119e06b4020492976b41034a80ce5.py, line 368, in render_body
  Module 66ab36ad7c0e2754627fffa33d83f359.py, line 138, in render_edit
  Module fb06dbb10037e040ae515369179628c1.py, line 665, in render_edit
  Module 66ab36ad7c0e2754627fffa33d83f359.py, line 100, in __fill_widget_body
  Module Products.Archetypes.Widget, line 956, in edit
  Module Products.Archetypes.Widget, line 1055, in _base_args
  Module Products.Archetypes.Widget, line 985, in _base_args
TypeError: archetypes.edit.beforefieldsets|nothing

 - Expression: "python:widget.edit(context, field, request)"
 - Filename:   ... Products/Archetypes/browser/templates/patterns_widget.pt
 - Location:   (line 19: col 36)
 - Source:     ... tructure python:widget.edit(context, field, request)"></div>
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 - Expression: "provider:archetypes.edit.beforefieldsets|nothing"
 - Filename:   ... cts/PloneFormGen/skins/PloneFormGen/fg_edit_macros_p3.pt
 - Location:   (line 92: col 38)
 - Source:     ... tructure provider:archetypes.edit.beforefieldsets|nothing"
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 - Arguments:  repeat: {...} (0)
               template: <ImplicitAcquisitionWrapper fg_base_view_p3 at 0x10f2d55f0>
               modules: <instance - at 0x1099a2bd8>
               portal_type: formfolder
               here: <ImplicitAcquisitionWrapper form at 0x10f2d5780>
               js: <list - at 0x110b55200>
               portal: <ImplicitAcquisitionWrapper Plone5en at 0x10f4020f0>
               user: <ImplicitAcquisitionWrapper - at 0x10f31b690>
               loop: {...} (2)
               nothing: <NoneType - at 0x108f3ae98>
               target_language: <NoneType - at 0x108f3ae98>
               container: <ImplicitAcquisitionWrapper Plone5en at 0x10f4020f0>
               default: <object - at 0x10903ebd0>
               fields: <list - at 0x110df3830>
               request: <instance - at 0x110df4680>
               wrapped_repeat: <SafeMapping - at 0x1118ee208>
               traverse_subpath: <list - at 0x10ffdce18>
               context: <ImplicitAcquisitionWrapper form at 0x10f2d5780>
               translate: <function translate at 0x1111d6500>
               root: <ImplicitAcquisitionWrapper Zope at 0x10f3a0b90>
               options: {...} (2)
               css: <list - at 0x110b55440>
```

**Comment by keul** [comment]
@mauritsvanrees I think I need to provide an upgrade step for migrating all fields to new ones

**Comment by mauritsvanrees** [comment]
I think so too. If you (or someone else) can do that, that would be awesome.

**Comment by keul** [comment]
@mauritsvanrees I've a customer interested on having PFG working on Plone 5 so he will probably be happy to let me work on this.

**Comment by keul** [comment]
@mauritsvanrees after a long time we pushed forward discussed changes and fixes. @gianniftp helped me finishing the migration step from old widget to new ones.

Let us know!

File: Products/PloneFormGen/content/fields.py [hunk 1]
```diff
@@ -505,20 +507,36 @@ def __init__(self, oid, **kwargs):
...
             )
 
-
     security.declareProtected(ModifyPortalContent, 'setFgShowHM')
     def setFgShowHM(self, value, **kw):
         """ set show_hm """
-
-        if type(value) == BooleanType:
-            self.fgField.widget.show_hm = value
+        if not type(value) == BooleanType:
```
Comment (mauritsvanrees) [inline Products/PloneFormGen/content/fields.py]: This should be `!=` instead of `not ==`. But we should have been using `isinstance` anyway.

**Comment by mauritsvanrees** [comment]
I have released this in 1.8.5. Thank you!

cc @smcmahon

**Pull request merged**